src/WordPress/JsonMapper/JsonMapper.php

```diff
@@ -0,0 +1,362 @@
+<?php
+
+namespace WordPress\JsonMapper;
+
+use ArrayObject;
+use ReflectionClass;
+use ReflectionException;
+use ReflectionMethod;
+use stdClass;
+/**
+ * Class JsonMapper
+ *
+ * This class is responsible for mapping JSON data to PHP objects. It supports mapping to native PHP classes
+ * as well as custom classes. Custom factories can be provided for instantiating objects of specific classes.
+ *
+ * @package WordPress\JsonMapper
+ */
+class JsonMapper {
+	const ARRAY_TYPE       = '/^array(\[])*$/';
+	const MIXED_ARRAY_TYPE = '/^mixed(\[])+$/';
+	/**
+	 * Array of strings representing valid scalar types.
+	 *
+	 * @var string[]
+	 */
+	private $scalar_types = array( 'string', 'bool', 'boolean', 'int', 'integer', 'double', 'float' );
+
+	/**
+	 * Array of factories for instantiating objects of specific classes.
+	 *
+	 * @var array<string,callable>
+	 */
+	private $factories = array();
+
+	/**
+	 * Constructs a new instance of this class.
+	 *
+	 * This constructor initializes the object and adds factories for native PHP classes.
+	 * It also allows for the addition of custom factories through the $custom_factories parameter.
+	 *
+	 * @param null|array $custom_factories An associative array where the key is the class name
+	 * and the value is a callable factory function. The factory function is expected to take
+	 * an instance of stdClass as its argument and return an instance of the class specified
+	 * by the key. This parameter is optional, and if not provided, an empty array will be used.
+	 */
+	public function __construct( array $custom_factories = array() ) {
+		$this->add_factories_for_native_php_classes();
+		$this->add_custom_factories( $custom_factories );
+	}
+
+	/**
+	 * Creates an instance of the given class and populates it with data from the given JSON object.
+	 *
+	 * This method first checks if a factory exists for the given class. If a factory exists, it uses the factory
```

Comment:
Is it GPT generated? That docstring isn't adding any context on top of the implementation and also has to kept in sync with the code changes. Let's remove all the docstring lines that are just english transcriptions of the code lines below them. A useful docstring says something about the method or usage that isn't immediately obvious otherwise.

Reply:
It was. But I've edited it heavily. I guess I was swayed by your inclination to make this library as beginner-friendly (any skillset dev friendly) as possible and I overdid it here. Thanks for your remarks. I've analyzed the DocBlocks you've committed, so it will be easier to hit closer to the target next time. Thanks for that!